src/ApiClient.php

```diff
@@ -39,17 +40,32 @@ class ApiClient
      */
     private $userAgent;
 
+    private CookieJar $cookieJar;
+    private $cookiesCacheFile = __DIR__.'/cookies.txt';
+    private $crumbCacheFile = __DIR__.'/crumb.txt';
+
     public function __construct(ClientInterface $guzzleClient, ResultDecoder $resultDecoder)
     {
         $this->client = $guzzleClient;
         $this->resultDecoder = $resultDecoder;
         $this->userAgent = UserAgent::getRandomUserAgent();
+        $this->client = new Client([
+            'headers' => [
+                'User-Agent' => $this->userAgent,
+                'Accept' => 'application/json, text/plain, */*',
+                'Accept-Language' => 'en-US,en;q=0.9',
+                'Referer' => 'https://finance.yahoo.com/',
+                'Origin' => 'https://finance.yahoo.com',
+            ],
+            'http_errors' => false,
+        ]);
```

Comment:
Multiple issues with this block:
- The client is overwritten immediately, making the `$guzzleClient` parameter useless. The client object that is passed should be used, rather than creating a new one.
- Making all HTTP errors "silent" (no exception) is a breaking change. And I believe it makes some of your code impossible to reach. For example, you have implemented `catch` statements for 401 `ClientException`, but these exceptions would never be thrown with `'http_errors' => false`